### PR TITLE
Add location Bigquery configuration and several fixes ['change:impactful`]

### PR DESCRIPTION
**Pull request by jmriego**
## Context

This will add several fixes to the BigQuery target that we have been using in production for quite some time now

### Changes

- Fix for reserved table names (allows table names such as `full`)
- Fix for bad decimals (fixes for when decimals in the source system have more digits than allowed in bigquery)
- Improved logging of bad data (shows the record that it's failing on the logs)
- Refactor code to make it more similar to other taps

## Checklist
- [x] I have considered the impact of this change and added a [Change Classification](
https://transferwise.atlassian.net/wiki/spaces/EKB/pages/1401189673/Change+Classifications+and+Expectations) Label (`change:standard`, `change:impactful`, `change:emergency`)
- [x] Change meets or does not compromise the [Baseline Security Requirements](https://transferwise.atlassian.net/wiki/spaces/EKB/pages/434929973/Baseline+Security+Requirements)

### CODE REVIEW

**Comment by jmriego** [comment]
fixed conflicts

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,17 @@
+1.1.0 (2021-05-21)
+-------------------
+- Add location BigQuery configuration
+- Fix for reserved table names
+- Fix for bad decimals
+- Improved logging of bad data
+- Refactor code to make it more similar to other taps
+
+1.0.2 (2020-07-16)
```
Comment (koszti) [inline CHANGELOG.md]: is 1.0.2 an existing version? I can't see it in master or in pypi

Comment (jmriego) [reply CHANGELOG.md]: oh not at all, I was using that in our internal repo. Good catch! I just removed it

I have also named the new version we could generate after merging this as 1.1.0 but let me know if you are following another standard

**Review by koszti — APPROVED** [review]

**Comment by koszti** [comment]
@jmriego CI is currently failing at pylint. Can you please fix it as another PR?

We should also move the content of .circle into .github as GHActions so you will also see the output of the tests when sending PR. Perhaps that's another task we'll need to do later. 

```
pylint target_bigquery -d C,W,unexpected-keyword-arg,duplicate-code
************* Module target_bigquery
target_bigquery/__init__.py:102: [E0602(undefined-variable), persist_lines] Undefined variable 'InvalidValidationOperationException'
target_bigquery/__init__.py:106: [E0602(undefined-variable), persist_lines] Undefined variable 'RecordValidationException'
************* Module target_bigquery.stream_utils
target_bigquery/stream_utils.py:97: [R1705(no-else-return), add_metadata_values_to_record.parse_datetime] Unnecessary "elif" after "return"

-----------------------------------
Your code has been rated at 9.84/10
```

**Comment by jmriego** [comment]
hi @koszti , no problem. I'll add those changes. Sorry if I missed something but how can I currently see the results of the linting?

No worries, just checked the circleci script

**Comment by koszti** [comment]
@jmriego follow the instructions at https://github.com/transferwise/pipelinewise-target-bigquery#to-run-pylint

**Comment by jmriego** [comment]
thanks @koszti ! I just sent a new merge request fixing those.
I have also bumped the version in setup.py. Is there anything else needed so the new version gets uploaded to PyPi?

**Comment by koszti** [comment]
thanks @jmriego . Lint is fixed but unit tests are failing:

```
#!/bin/bash -eo pipefail
. venv/bin/activate
export LOGGING_CONF_FILE=$(pwd)/sample_logging.conf
nosetests --where=tests/unit
.....Etime=2021-06-04 08:18:28 name=target_bigquery level=INFO message=Emitting state "{\"currently_syncing\": null}"
.
======================================================================
ERROR: test_adjust_timestamps_in_record (test_target_bigquery.TestTargetBigQuery)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/circleci/project/tests/unit/test_target_bigquery.py", line 74, in test_adjust_timestamps_in_record
    target_bigquery.adjust_timestamps_in_record(record, schema)
AttributeError: module 'target_bigquery' has no attribute 'adjust_timestamps_in_record'
```

Can you please fix it and send another PR?

Please don't bump the version, we'll cut the version and update the Changelog separately once everything is working.

**Comment by jmriego** [comment]
Sorry about that. I have fixed that and checked again all unit tests, integration tests and linting. Pushed a new PR following this. Thanks!

**Pull request merged**